### PR TITLE
Fixes related to test mode prompt style and copy

### LINKED ISSUE #6778: [PO test mode] Copy should be "Set up" in two words and update to CTA
### Description
In test mode for Progressive Onboarding, the grammar is incorrect in the test mode flow on the Payments Overview. "Setup" should be two words - "Set up" in the Overview and modal. "Set up" is the verb form, and "Setup" is the noun, which is why both appear in different contexts. :D

Also, for better comprehension on the page and consistency with other areas of Woo, the CTA should be aligned left in the card footer and secondary style when appearing inline on the page. However, modals in Woo all continue to have the right aligned CTA.

### Designs

On Payments Overview in Progressive Onboarding test mode
image

Modal that appears after clicking "Set up payments"
image

cc @dmallory42

### PR DESCRIPTION
**Pull request by dmallory42**
Fixes #6778

#### Changes proposed in this Pull Request

* Fixes to the styling and copy of the "Set Up Real Payments" option on the account page and modal.

### Overview Page 

**Before:** 

Screenshot 2023-07-24 at 14 22 53

**After:**

Screenshot 2023-07-24 at 14 21 24

### Modal

**Before:**

Screenshot 2023-07-24 at 14 22 57

**After:**

Screenshot 2023-07-24 at 14 21 38

#### Testing instructions

* Check out this branch on your client.
* Update the `wcpay_accounts` table in `localhost:8087` to set the `account_type` field to `wcpay_old` (or similar) for any active accounts. (To trick the server into thinking no account is connected)
* Refresh the cache on your local store. Ensure `Enable Progressive Onboarding` is enabled.
* Onboard a new account, select the builder path to create a test account.
* After onboarding is finished, you should be taken to the `Payments > Overview` page.
* Confirm that the style and copy of the content in the `Set up real payments` section matches the designs in #6830. 

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6830` or branch name `dev/6778-fix-copy-test-mode` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=dev/6778-fix-copy-test-mode) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: def6a83056cbc81956151ae55dd9ff5691507e54
- Build time: 2023-08-02 09:26:30 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by dmallory42** [comment]
cc. @elizaan36 for design review, please. 🙂

**Comment by elizaan36** [comment]
Thanks @dmallory42, looks good to me!

One question - We have a shared component via Gutenberg for the modal. Is that being used here? I'm asking because most modals have the header with bottom border. I know this implementation matches the design exactly so this is just out of curiosity.

File: client/overview/setup-real-payments.tsx [hunk 1]
```diff
@@ -63,15 +64,20 @@ const SetupRealPayments: React.FC = () => {
 					</div>
 				</CardBody>
 				<CardFooter className="wcpay-setup-real-payments__footer">
-					<Button isPrimary onClick={ () => setModalVisible( true ) }>
-						{ __( 'Set up payments', 'woocommerce-payments' ) }
-					</Button>
+					<Flex align="center" justify="flex-start">
+						<Button
+							isSecondary={ true }
```
Comment (oaratovskyi) [inline client/overview/setup-real-payments.tsx]: I see that this attribute is marked as deprecated 
image
We could use `variant={ 'secondary' }`, I see it's used in our codebase, e.g. [here](https://github.com/Automattic/woocommerce-payments/blob/d13d3268f26ce9238fd24f315e30c8332b4e800c/client/disputes/index.tsx#L301-L301)

**Review by oaratovskyi — APPROVED** [review]
LGTM and works well! :shipit: 
Have only one nitpick 🙂

File: client/overview/setup-real-payments.tsx (hunk 1 above)
Comment (dmallory42) [reply client/overview/setup-real-payments.tsx]: Ah, thank you! I noticed that too but I wasn't sure what the alternative was. Will make that adjustment before merge! 🙂

Comment (dmallory42) [reply client/overview/setup-real-payments.tsx]: Addressed in 8963da319.

**Comment by dmallory42** [comment]
Thanks for the design review, @elizaan36! And thanks for the code review, @oaratovskyi. 🙂

> One question - We have a shared component via Gutenberg for the modal. Is that being used here? I'm asking because most modals have the header with bottom border. I know this implementation matches the design exactly so this is just out of curiosity.

Yes, it is using the `Modal` component from Gutenberg. I think the reason that bottom border under the heading isn't here is because we aren't using the `heading` element that the modal would normally have (I'm not sure why, my guess is just to match the design). 🙂 Let me know if you prefer it with the border and we can adjust it.

**Comment by elizaan36** [comment]
Thanks @dmallory42 ! If there's the possibility to use the header bottom border that would be great for consistency across other Woo modals. Just mocked up quickly what that would look like. Sorry for the oversight!

Screen Shot 2023-07-26 at 8 28 27 AM

**Comment by dmallory42** [comment]
Hey @elizaan36! I took another look at the [Gutenberg Modal](https://developer.wordpress.org/block-editor/reference-guides/components/modal/) component and it looks like the default behaviour is not to have a header bottom border. 

I think maybe this changed at some point, so perhaps the WP component had a border in the past - I can definitely add some custom style to match the design you shared, but it may be better to move that to another issue since I can do it by creating a custom component style based on the Gutenberg modal. But my preference is probably to stick as close to WP as possible here. 

Let me know what you think :)

**Comment by elizaan36** [comment]
Yeah the component was just changed! Here's the new one with no bottom border https://wordpress.github.io/gutenberg/?path=/story/components-modal--default

No worries on adding the border since like you said we'll want to start aligning Woo with the new component sans customization.

**Pull request merged**